### DISCUSSION TITLE
removing a component inside a member function of this component

### DISCUSSION THREAD

**Discussion by Nathan-M-code** [skypjack/entt#981] [Q&A] [answered]
Hello,
Sorry if it is a easy/dumb question, I am new with ecs and entt.
My component is a class, is there any trouble if I use `reg.remove<Compo>(m_entity);` inside a member function of Compo ?
If no, will the instance of Compo be destroyed ? When ?
Thank you

**Comment by skypjack** [comment, marked as answer]
If the component is assigned to `m_entity`, then yes, it's destroyed.
Is it safe? According to C++, yes. It's not different from doing a `delete this`, which is perfectly fine in C++ if you know what you're doing. 🤷‍♂️ 
On the other hand, if the question is _am I going to hurt myself_? Well, I don't know 🙂 since I don't have access to the full codebase nor the time to inspect it to give you a proper answer.

**Reply by Nathan-M-code** [reply]
Thank you, that answers my question !
